ext/shopify-extensions/shopify_extensions.rb

```diff
@@ -0,0 +1,198 @@
+require "rbconfig"
+require "open-uri"
+require "json"
+require "zlib"
+require "rubygems"
+require "rubygems/package"
+require "open3"
+
+module ShopifyExtensions
+  def self.install(**args)
+    Install.call(**args)
+  end
+
+  class Install
+    def self.call(platform: Platform.new, **args)
+      new.call(platform: platform, **args)
+    end
+
+    def call(platform:, version:, target:)
+      target = platform.format_path(target)
+
+      release = fetch_release_details_for(version: version)
+      raise InstallationError.version_not_found(version) unless release
+
+      downloaded = release.download(platform: platform, target: target)
+      raise InstallationError.asset_not_found(platform: platform, version: version) unless downloaded
+
+      raise InstallationError.installation_failed unless verify(target, version: version)
+    end
+
+    private
+
+    def fetch_release_details_for(version:)
+      JSON.parse(URI.parse(release_url_for(version: version)).open.read).yield_self(&Release)
+    rescue OpenURI::HTTPError
+      nil
+    end
+
+    def release_url_for(version:)
+      format(
+        "https://api.github.com/repos/%{owner}/%{repo}/releases/tags/%{version}",
```

Comment:
That's definitely easier and I will explore that. That said, does Github enforce a global API limit on these endpoints or are limits bound by IP? Given that the requests are coming from various IPs would we even run into limitations?

Reply:
> That said and more out of general curiosity, does Github enforce a global API limit on these endpoints or are limits bound by IP? Given that the requests are coming from various IPs would we even run into limitations?

I don't think those API limits are documented but I can tell this became quite problematic and had to revert changes in the past. My guess is that it's IP-bound because it started failing a lot in CI environments. One way I've seen other projects workaround this is by reading a GitHub token either from the `git` cli or the CI environment.